### PR TITLE
Hide 'My selections' filter option for variants in display only view

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**
Closes [CXSPA-7354](https://jira.tools.sap/browse/CXSPA-7354)

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html [hunk 1]
```diff
@@ -28,28 +28,61 @@
     >
   </div>
 </div>
+<ng-container
+  *ngIf="!_config.features?.productConfiguratorHideMySelectionsFilterOptions"
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html]: I am not 100% sure we need this feature toggle here (because isn't it rather a bug fix to not display my selections for variants?). We are also labelling the whole activity as bug fix

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts [hunk 2]
```diff
@@ -14,6 +22,10 @@ import { Configurator } from '../../core/model/configurator.model';
   templateUrl: './configurator-overview-filter.component.html',
 })
 export class ConfiguratorOverviewFilterComponent implements OnChanges {
+  protected productService = inject(ProductService);
+  protected routingService = inject(RoutingService);
+  protected _config = inject(Config);
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts]: Please don't use '_' here (because that's typically indicating that a variable is not used)

Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts]: Anyhow I guess it's not needed, let's discuss

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts [hunk 3]
```diff
@@ -44,6 +56,35 @@ export class ConfiguratorOverviewFilterComponent implements OnChanges {
...
+        switchMap((routerState) => {
+          return routerState.state.queryParams.productCode
+            ? this.productService.get(
+                routerState.state.queryParams.productCode,
+                ProductScope.LIST
+              )
+            : EMPTY;
+        })
+      )
+      .pipe(
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts]: Not needed, you are already piping in line #70

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html [hunk 4]
```diff
@@ -28,28 +28,61 @@
...
-      for="cx-configurator-overview-filter-option-mySelections"
-    >
-      {{ 'configurator.overviewFilter.mySelections' | cxTranslate }}</label
-    >
+<ng-container
+  *ngIf="_config.features?.productConfiguratorHideMySelectionsFilterOptions"
+>
+  <div
+    *ngIf="!(isDisplayOnlyVariant() | async)"
+    class="cx-overview-filter-option"
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html]: We should also remove the 'Price relevant options' filter then, shouldn't we? 
Because we cannot tell this for a variant either

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts [hunk 5]
```diff
@@ -44,6 +56,35 @@ export class ConfiguratorOverviewFilterComponent implements OnChanges {
     this.configuratorCommonsService.updateConfigurationOverview(inputConfig);
   }
 
+  /**
+   * Verifies whether a product is a variant product in the display only view.
+   *
+   * @returns - if `baseProduct` property of the current product is defined
+   * and provides the product code of the base product,
+   * and the current product is in the display only view
+   * then returns `true`, otherwise `false`.
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts:59]: The JS doc is not entirely true. Because for editable configurations, the method will not return false, it will never emit. So either change the method (which would be probably better understandable) or adopt the comment.

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts [hunk 6]
```diff
@@ -44,6 +56,35 @@ export class ConfiguratorOverviewFilterComponent implements OnChanges {
...
+   */
+  isDisplayOnlyVariant(): Observable<boolean> {
+    return this.routingService
+      .getRouterState()
+      .pipe(
+        filter((routerState) => routerState.state.params.displayOnly),
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts]: Can we do without the filter but rather emit false? See previous comment

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.spec.ts [hunk 7]
```diff
@@ -72,6 +78,20 @@ class MockConfiguratorStorefrontUtilsService {
   getVerticallyScrolledPixels(): void {}
 }
 
+class MockRoutingService {
+  getRouterState(): Observable<RouterState> {
+    return of(ConfigurationTestData.mockRouterState);
+  }
+
+  go = () => Promise.resolve(true);
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.spec.ts:87]: Why do we need to use Promise here?

File: projects/core/src/features-config/feature-toggles/config/feature-toggles.ts [hunk 8]
```diff
@@ -245,6 +245,12 @@ export interface FeatureTogglesInterface {
    */
   a11yEmptyWishlistHeading?: boolean;
 
+  /**
+   * In `ConfiguratorOverviewFilterComponent` it hides `My Selections` filter option on the overview page in case
+   * the current product is a variant in the display only view.
+   */
+  productConfiguratorHideMySelectionsFilterOptions?: boolean;
```
Comment (ChristophHi) [inline projects/core/src/features-config/feature-toggles/config/feature-toggles.ts]: Probably not needed, see previous comment

File: projects/storefrontapp-e2e-cypress/cypress/e2e/regression/product_configurator/product-configurator-textfield.e2e.cy.ts [hunk 9]
```diff
@@ -18,7 +18,7 @@ context('Textfield Configuration', () => {
   before(() => {
     cy.visit('/');
   });
-  describe('Navigate to Textfield Configuration Page', () => {
+  describe('Navigate to textfield configuration page', () => {
```
Comment (ChristophHi) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/regression/product_configurator/product-configurator-textfield.e2e.cy.ts]: Please revert as this has nothing to do with our fix

File: projects/storefrontapp-e2e-cypress/cypress/e2e/regression/product_configurator/product-configurator-vc-cart.e2e-flaky.cy.ts [hunk 10]
```diff
@@ -84,7 +84,7 @@ context('Product Configuration', () => {
     configUISettings.productConfigurator.enableNavigationToConflict = false;
   });
 
-  describe('Navigate to Product Configuration Page', () => {
+  describe('Navigate to product configuration page', () => {
```
Comment (ChristophHi) [inline projects/storefrontapp-e2e-cypress/cypress/e2e/regression/product_configurator/product-configurator-vc-cart.e2e-flaky.cy.ts]: Better do these changes in a separate PR (that would be of type test: rather than fix: )

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html]: Agree

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.spec.ts [hunk 11]
```diff
@@ -101,10 +151,41 @@ describe('ConfiguratorOverviewFilterComponent', () => {
...
+      fixture.detectChanges();
+
+      CommonConfiguratorTestUtilsService.expectElementPresent(
+        expect,
+        htmlElem,
+        '.cx-no-overview-filters-available'
+      );
+    });
 
+    it('should render always default options', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.spec.ts]:
```suggestion
    it('should always render default options', () => {
```

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.spec.ts [hunk 12]
```diff
@@ -113,8 +194,23 @@ describe('ConfiguratorOverviewFilterComponent', () => {
       );
     });
 
+    it('should not render any filter options', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.spec.ts]:
```suggestion
    it('should not render any filter options for read-only variants', () => {
```

**Review by ChristophHi — CHANGES_REQUESTED** [review]

**Review by ChristophHi — DISMISSED** [review]

**Review by ChristophHi — APPROVED** [review]
